### ISSUE TITLE
[Feature request] Support for typing hints

### ISSUE THREAD

**Issue by pjljvandelaar** [neo4j/neo4j-python-driver#493]
I would like to use the full capability of [mypy](http://mypy-lang.org/) while using neo4j.

However, neo4j driver is not annotated for that purpose.
One gets
```
example.py:4: error: Skipping analyzing 'neo4j': found module but no type hints or library stubs
example.py:4: note: See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports
```
Adding `# type: ignore`  after the import `from neo4j import GraphDatabase` as suggested on the provided html page only partly solves the problem.
I no longer am able to use the flag `--disallow-any-expr` on my complete archive.
On all neo4j related lines I now get:
```
example.py:17: error: Expression has type "Any"
example.py:19: error: Expression has type "Any"
example.py:22: error: Expression has type "Any"
example.py:30: error: Expression has type "Any"
example.py:31: error: Expression has type "Any"
example.py:33: error: Expression has type "Any"
example.py:35: error: Expression has type "Any"
```

So please add type annotations to the neo4j driver: I expect that it will prevent many subtle error in my and other user's code.
For me working with untyped graphs feels like walking a tight rope without a safety net!

**Comment by robsdedude** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Add type hints](https://trello.com/c/rt4LYz4R/647-add-type-hints)

**Comment by robsdedude** [comment]
Thanks for the suggestion. I've added the idea to our internal Trello wall. I can't promise that it will happen, though. Definitely not in the near future as other things have higher priority right now.

**Comment by robsdedude** [comment]
Good news. The first draft of type hints will be released soon in version 5.0.0a2

https://github.com/neo4j/neo4j-python-driver/pull/767
